### PR TITLE
Add test for DNS options set from kickstart

### PR DESCRIPTION
**Pull request by VladimirSlavik**
This tests https://github.com/rhinstaller/anaconda/pull/4519.

### CODE REVIEW

**Comment by VladimirSlavik** [comment]
The actual code to be tested is buggy (as of right now), so the test cannot succeed.

There is still more work to be done on the test - apart from the bugs that are straightforward to explain, the test sees the correct values on connections, but not in the config files. Which brings up some more questions...

**Comment by VladimirSlavik** [comment]
So, it seems that the checks for actual device config pass only for the installer environment, ie. running the checks in `%post --nochroot`. However, I am not sure if that is correct? If not, that means something does not copy to the installed system, right?

If I leave it as a normal `%post`, I get:

```
*** Failed check: DOMAIN=fedoraproject.org,example.com in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv4.dns-search=fedoraproject.org,example.com in /etc/NetworkManager/system-connections/enp1s0.nmconnection
*** Failed check: IPV6_DOMAIN=whatever.nonexistent in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv6.dns-search=whatever.nonexistent in /etc/NetworkManager/system-connections/enp1s0.nmconnection
*** Failed check: PEERDNS=__NONE in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv4.ignore-auto-dns=no in /etc/NetworkManager/system-connections/enp1s0.nmconnection
*** Failed check: IPV6_PEERDNS=yes in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv6.ignore-auto-dns=yes in /etc/NetworkManager/system-connections/enp1s0.nmconnection
```

**Comment by rvykydal** [comment]
> So, it seems that the checks for actual device config pass only for the installer environment, ie. running the checks in `%post --nochroot`. However, I am not sure if that is correct? If not, that means something does not copy to the installed system, right?
> 
> If I leave it as a normal `%post`, I get:
> 
> ```
> *** Failed check: DOMAIN=fedoraproject.org,example.com in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv4.dns-search=fedoraproject.org,example.com in /etc/NetworkManager/system-connections/enp1s0.nmconnection
> *** Failed check: IPV6_DOMAIN=whatever.nonexistent in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv6.dns-search=whatever.nonexistent in /etc/NetworkManager/system-connections/enp1s0.nmconnection
> *** Failed check: PEERDNS=__NONE in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv4.ignore-auto-dns=no in /etc/NetworkManager/system-connections/enp1s0.nmconnection
> *** Failed check: IPV6_PEERDNS=yes in /etc/sysconfig/network-scripts/ifcfg-enp1s0 or ipv6.ignore-auto-dns=yes in /etc/NetworkManager/system-connections/enp1s0.nmconnection
> ```

If you look in the /tmp/syslog you can see this dump of the configuration file:
```
18:55:24,984 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:/mnt/sysroot/etc/NetworkManager/system-connections/enp1s0.nmconnection:
18:55:24,984 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[connection]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:id=enp1s0
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:uuid=30f27f93-b8c6-4bfd-a77e-2eb6615c5d83
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:type=ethernet
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:autoconnect-priority=-100
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:autoconnect-retries=1
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:interface-name=enp1s0
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:multi-connect=1
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:timestamp=1678992650
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:wait-device-timeout=60000
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[ethernet]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[ipv4]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:dns-search=fedoraproject.org;example.com;
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:method=auto
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[ipv6]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:addr-gen-mode=eui64
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:dns-search=whatever.nonexistent;
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:ignore-auto-dns=true
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:method=auto
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[proxy]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:[user]
18:55:24,985 WARNING org.fedoraproject.Anaconda.Modules.Network:DEBUG:anaconda.modules.network.network:org.freedesktop.NetworkManager.origin=nm-initrd-generator
```
So the checks  for keyfile format should be (the quotes are important):
```
check_device_config_value @KSTEST_NETDEV1@ DOMAIN fedoraproject.org,example.com ipv4 dns-search "fedoraproject.org;example.com;"
check_device_config_value @KSTEST_NETDEV1@ IPV6_DOMAIN whatever.nonexistent ipv6 dns-search "whatever.nonexistent;"

check_device_config_value @KSTEST_NETDEV1@ PEERDNS __NONE ipv4 ignore-auto-dns __NONE
check_device_config_value @KSTEST_NETDEV1@ IPV6_PEERDNS yes ipv6 ignore-auto-dns true
```

For the ifcfg file values you can look at RHEL8, where ifcfg format is default, or set ifcfg default for rawhide in NM configuration via updates image (or in dracut shell), like described in https://fedoraproject.org/wiki/Changes/NetworkManager_keyfile_instead_of_ifcfg_rh, ie configre "main.plugins=ifcfg-rh,keyfile"

File: network-dns-search.ks.in [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#version=DEVEL
+#test name: network-dns-search
+# Test domain name search properties set from kickstart
+
+# Use defaults.
+%ksappend repos/default.ks
+%ksappend common/common_no_payload.ks
```
Comment (rvykydal) [inline network-dns-search.ks.in:7]: This includes
```
network --bootproto=dhcp
```
which is not wanted for a networking test setting the network command, instead, I'd use this ksappend battery:
```
%ksappend common/common.ks
%ksappend repos/default.ks
%ksappend l10n/default.ks
%ksappend users/default.ks
%ksappend storage/default.ks
%ksappend payload/default_packages.ks
```

**Comment by VladimirSlavik** [comment]
Thank you for the tips! With these, I also managed to navigate the format zoo, so this should be it.

The test is disabled on rhel for now, since it would fail until the feature is ported and merged.

File: network-dns-search.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
```
Comment (rvykydal) [inline network-dns-search.sh:2]: 2023

File: network-dns-search.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Radek Vykydal <rvykydal@redhat.com>
```
Comment (rvykydal) [inline network-dns-search.sh:18]: you :)

**Comment by VladimirSlavik** [comment]
Waiting for a boot.iso with all the fixes merged.

**Comment by VladimirSlavik** [comment]
We are still waiting for a recent enough boot.iso. The test passes locally.

File: network-dns-search.ks.in [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+%post
+
+@KSINCLUDE@ post-lib-network.sh
+
+check_device_connected "@KSTEST_NETDEV1@" yes
+check_connection_device "@KSTEST_NETDEV1@" "@KSTEST_NETDEV1@"
+
+check_connection_setting "@KSTEST_NETDEV1@" ipv4.dns-search fedoraproject.org,example.com
+check_connection_setting "@KSTEST_NETDEV1@" ipv6.dns-search whatever.nonexistent
+check_device_config_value "@KSTEST_NETDEV1@" DOMAIN fedoraproject.org,example.com ipv4 dns-search "fedoraproject.org;example.com;"
```
Comment (VladimirSlavik) [inline network-dns-search.ks.in:25]: Just found out now: The ifcfg format uses space separators, not commas.

Comment (VladimirSlavik) [reply network-dns-search.ks.in:25]: Fixing that in #947.

**Pull request merged**